components/Container/index.tsx

```diff
@@ -0,0 +1,292 @@
+    </>
+  )
+}
+
+
+/**
+ * A toolbar that will reside inside a layout container.
+ */
+export const ToolbarContainer: React.FC<LayoutContainerProps> = ({
+  children,
+  style,
+  id = '',
+  className = '',
+}) => {
+  return (
+    <>
+      <div
+        id={id}
+        className={`layout-container layout-toolbar-container ${className}`}
+        style={style}
+      >
+        {children}
+      </div>
+    </>
+  )
+}
+
+
+/**
+ * Layout with a dynamic sidebar
+ */
+interface LayoutWithSidebarProps extends LayoutContainerProps {
+}
+
+export const LayoutWithSidebar: React.FC<LayoutWithSidebarProps> = ({
+  children,
+  id,
+  className = "",
+}) => {
+  const thisRef = useRef<HTMLDivElement | null>(null);
+
+  useEffect(() => {
+    if (thisRef) {
+      if (thisRef.current) {
+        let direction: DirectionType = null;
+
+        children?.forEach((rNode) => {
+          if (!rNode) return;
+
+          // Set up the state with a toolbar if present
+          switch (rNode.type.name) {
+            case "ToobarContainer": {
+              thisRef.current?.classList.add('with-toolbar');
+              break;
+            }
+
+            case "ContentContainer": {
+              if (!direction) direction = "right";
+            }
+
+            case "SidebarContainer": {
+              thisRef.current?.classList.add('with-sidebar');
+              if (!direction) direction = "left";
+              break;
+            }
+
+            case "DrawerContainer": {
+              thisRef.current?.classList.add('with-drawer');
+              if (!direction) direction = "left";
+            }
+          }
+        });
+
+        thisRef.current.classList.add(`direction-${direction}`);
+      }
+    }
+  }, []);
+
+  return (
+    <>
+      <div
+        ref={thisRef}
+        id={id}
+        className={`layout-container layout-with-panel ${className}`}
+      >
+        {children}
+      </div>
+    </>
+  )
+}
+
+interface SidebarContainerProps extends ContentContainerProps {
+  isOpen?: Boolean;
+}
+
+export const SidebarContainer: React.FC<SidebarContainerProps> = ({
+  children,
+  id = '',
+  className = '',
+
+  /**
+   * Holds the current state of the sidebar.
+   */
+  isOpen = true,
+}) => {
+  return (
+    <>
+      <div
+        id={id}
+        className={`layout-sidebar-container ${className} ${isOpen ? "state-open" : "state-closed"}`}
+      >
+        {children}
+      </div>
+    </>
+  );
+}
+
+// Drawer Layout Container
+interface DrawerContainerProps extends ContentContainerProps {
+  isOpen?: Boolean;
+  onClose?: () => void;
+}
+
+export const DrawerContainer: React.FC<DrawerContainerProps> = ({
+  children,
+  id = '',
+  className = '',
+  isOpen = false,
+  onClose,
+}) => {
+
+  const drawerRef = useRef<HTMLDivElement | null>(null);
+  const size = useResponsive();
+  const [isMobile, setIsMobile] = useState<Boolean>(true);
+  const [prevFocus, setPrevFocus] = useState<Element | null>(null);
+
+  useEffect(() => {
```

Comment:
L219: This can be simplified to: 
`useEffect(() => { setIsMobile(size.viewport[0] < getSizeByName('md')[0]); }, [size]);`